docs/proposals/control-plane-config.md

```diff
@@ -17,3 +17,86 @@ option that we will support is log level.
 ## Non-Goals
 
 This proposal is *not* defining a way to dynamically configure the data plane.
+
+## Introduction
+
+The NKG control plane will evolve to have various user-configurable options. These could include, but are not
+limited to, log level, tracing, or metrics. For the best user experience, these options should be able to be
+changed at runtime, to avoid having to restart NKG. The first option that we will allow users to configure is the
+log level. The easiest and most intuitive way to implement a Kubernetes-native API is through a CRD.
+
+## API, Customer Driven Interfaces, and User Experience
+
+The API would be provided in a CRD. An authorized user would interact with this CRD using `kubectl` to `get`
+or `edit` the configuration.
+
+Proposed configuration CRD example:
+
+```yaml
+apiVersion: nginx.gateway.k8s.io/v1beta1
+kind: NGINXControlConfig
```

Comment:
is having NGINX in the kind can limit us in the future if we start supporting more data planes?

Reply:
Yeah, I was going back and forth on this one. We could call it something like `ControlConfig`, but I liked having nginx in the name which would make it easy to find for a user. While the API group links it to us, most of the time when users look up a resource in k8s, they just look up the type. `ControlConfig` might seem too generic? And I see NGINX as representing the organization, not the data plane name in this case.

Reply:
I'm open to suggestions though.

Reply:
> And I see NGINX as representing the organization, not the data plane name in this case

Same. I think we should keep NGINX in the name.